File: superstitioMain/java/superstitio/cards/general/SkillCard/energyAndDraw/EscapeConjuring.java
package superstitio.cards.general.SkillCard.energyAndDraw;

import basemod.cardmods.ExhaustMod;
import basemod.helpers.CardModifierManager;
import com.megacrit.cardcrawl.actions.common.DiscardSpecificCardAction;
import com.megacrit.cardcrawl.actions.common.GainEnergyAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import com.megacrit.cardcrawl.powers.AbstractPower;
import superstitio.DataManager;
import superstitio.cards.general.GeneralCard;
import superstitio.powers.SexualHeat;

public class EscapeConjuring extends GeneralCard {
    public static final String ID = DataManager.MakeTextID(EscapeConjuring.class);

    public static final CardType CARD_TYPE = CardType.SKILL;

    public static final CardRarity CARD_RARITY = CardRarity.UNCOMMON;

    public static final CardTarget CARD_TARGET = CardTarget.SELF;

    private static final int COST = 0;
    private static final int MAGIC = 3;
    private static final int UPGRADE_MAGIC = 1;


    public EscapeConjuring() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        CardModifierManager.addModifier(this, new ExhaustMod());
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC);
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        int discardCard = (int) AbstractDungeon.player.hand.group.stream()
                .filter(card -> card.costForTurn != 0 && !card.freeToPlay()).count();
        int statusCardNum = (int) AbstractDungeon.player.hand.group.stream()
                .filter(card -> card.type == CardType.STATUS || card.type == CardType.CURSE).count();
        int deBuffNum = (int) AbstractDungeon.player.powers.stream()
                .filter(power -> power.type == AbstractPower.PowerType.DEBUFF).count();

        SexualHeat.addToBot_addSexualHeat(AbstractDungeon.player, deBuffNum * this.magicNumber);
        addToBot(new GainEnergyAction(statusCardNum));

        AbstractDungeon.player.hand.group.stream().filter(card -> card.costForTurn != 0 && !card.freeToPlay())
                .map(DiscardSpecificCardAction::new).forEach(this::addToBot);
        addToBot_drawCards(statusCardNum);
    }

    @Override
    public void upgradeAuto() {
        CardModifierManager.removeModifiersById(this, ExhaustMod.ID, false);
    }
}

File: superstitioMain/java/superstitio/cards/general/SkillCard/energyAndDraw/EscapeConjuring.kt
package superstitio.cards.general.SkillCard.energyAndDraw

import basemod.cardmods.ExhaustMod
import basemod.helpers.CardModifierManager
import com.megacrit.cardcrawl.actions.common.DiscardSpecificCardAction
import com.megacrit.cardcrawl.actions.common.GainEnergyAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster
import com.megacrit.cardcrawl.powers.AbstractPower
import com.megacrit.cardcrawl.powers.AbstractPower.PowerType
import superstitio.DataManager
import superstitio.cards.general.GeneralCard
import superstitio.powers.SexualHeat

class EscapeConjuring : GeneralCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET) {
    init {
        CardModifierManager.addModifier(this, ExhaustMod())
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC)
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?) {
        val discardCard = AbstractDungeon.player.hand.group.stream()
            .filter { card: AbstractCard -> card.costForTurn != 0 && !card.freeToPlay() }.count().toInt()
        val statusCardNum = AbstractDungeon.player.hand.group.stream()
            .filter { card: AbstractCard -> card.type == CardType.STATUS || card.type == CardType.CURSE }.count()
            .toInt()
        val deBuffNum = AbstractDungeon.player.powers.stream()
            .filter { power: AbstractPower -> power.type == PowerType.DEBUFF }.count().toInt()

        SexualHeat.addToBot_addSexualHeat(AbstractDungeon.player, deBuffNum * this.magicNumber)

        //        addToBot(new GainEnergyAction(statusCardNum));
        AbstractDungeon.player.hand.group.stream()
            .filter { card: AbstractCard -> card.type == CardType.STATUS || card.type == CardType.CURSE }
            .map(::DiscardSpecificCardAction)
            .forEach(this::addToBot)
        addToBot(GainEnergyAction(statusCardNum))
        addToBot_drawCards(statusCardNum)
    }

    override fun upgradeAuto() {
        CardModifierManager.removeModifiersById(this, ExhaustMod.ID, false)
    }

    companion object {
        val ID: String = DataManager.MakeTextID(EscapeConjuring::class.java)

        val CARD_TYPE: CardType = CardType.SKILL

        val CARD_RARITY: CardRarity = CardRarity.UNCOMMON

        val CARD_TARGET: CardTarget = CardTarget.SELF

        private const val COST = 0
        private const val MAGIC = 3
        private const val UPGRADE_MAGIC = 1
    }
}
